Dependency graph is throwing error with Mysql storage

Zipkin Version: 1.25.3

Dependency graph is throwing errors when used with Mysql storage. Stack-trace of zipkin-query is:
```
23:39:45.234 [pool-1-thread-2] ERROR c.t.f.h.i.e.FinatraDefaultExceptionMapper - internal server error
java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')
AND a_key in ("sr","sa")
AND endpoint_service_name is not null
GROUP BY trace_' at line 1
Query is:
SELECT DISTINCT trace_id, span_id, endpoint_service_name
FROM zipkin_annotations
WHERE trace_id IN ()
AND a_key in ("sr","sa")
AND endpoint_service_name is not null
GROUP BY trace_id, span_id
        
	at org.mariadb.jdbc.internal.util.ExceptionMapper.get(ExceptionMapper.java:125) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.internal.util.ExceptionMapper.throwException(ExceptionMapper.java:69) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.MariaDbStatement.executeQueryEpilog(MariaDbStatement.java:261) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:290) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.MariaDbStatement.executeQuery(MariaDbStatement.java:448) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.MariaDbClientPreparedStatement.executeQuery(MariaDbClientPreparedStatement.java:119) ~[zipkin-query.jar:na]
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52) ~[zipkin-query.jar:na]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java) ~[zipkin-query.jar:na]
	at anorm.Sql$$anonfun$resultSet$1$$anonfun$apply$3.apply(Anorm.scala:140) ~[zipkin-query.jar:na]
	at anorm.Sql$$anonfun$resultSet$1$$anonfun$apply$3.apply(Anorm.scala:140) ~[zipkin-query.jar:na]
	at resource.DefaultManagedResource.open(AbstractManagedResource.scala:106) ~[zipkin-query.jar:na]
	at resource.AbstractManagedResource.acquireFor(AbstractManagedResource.scala:85) ~[zipkin-query.jar:na]
	at resource.ManagedResourceOperations$$anon$2$$anonfun$acquireFor$1.apply(ManagedResourceOperations.scala:40) ~[zipkin-query.jar:na]
	at resource.ManagedResourceOperations$$anon$2$$anonfun$acquireFor$1.apply(ManagedResourceOperations.scala:40) ~[zipkin-query.jar:na]
	at resource.AbstractManagedResource$$anonfun$5.apply(AbstractManagedResource.scala:86) ~[zipkin-query.jar:na]
	at scala.util.control.Exception$Catch$$anonfun$either$1.apply(Exception.scala:125) ~[zipkin-query.jar:na]
	at scala.util.control.Exception$Catch$$anonfun$either$1.apply(Exception.scala:125) ~[zipkin-query.jar:na]
	at scala.util.control.Exception$Catch.apply(Exception.scala:103) ~[zipkin-query.jar:na]
	at scala.util.control.Exception$Catch.either(Exception.scala:125) ~[zipkin-query.jar:na]
	at resource.AbstractManagedResource.acquireFor(AbstractManagedResource.scala:86) ~[zipkin-query.jar:na]
	at resource.ManagedResourceOperations$$anon$2.acquireFor(ManagedResourceOperations.scala:40) ~[zipkin-query.jar:na]
	at resource.DeferredExtractableManagedResource.acquireFor(AbstractManagedResource.scala:27) ~[zipkin-query.jar:na]
	at resource.ManagedResourceOperations$class.acquireAndGet(ManagedResourceOperations.scala:25) ~[zipkin-query.jar:na]
	at resource.DeferredExtractableManagedResource.acquireAndGet(AbstractManagedResource.scala:24) ~[zipkin-query.jar:na]
	at anorm.Sql$$anonfun$asTry$1.apply(Anorm.scala:215) ~[zipkin-query.jar:na]
	at anorm.Sql$$anonfun$asTry$1.apply(Anorm.scala:215) ~[zipkin-query.jar:na]
	at scala.util.Try$.apply(Try.scala:192) ~[zipkin-query.jar:na]
	at anorm.Sql$.asTry(Anorm.scala:215) ~[zipkin-query.jar:na]
	at anorm.WithResult$class.as(SqlResult.scala:120) ~[zipkin-query.jar:na]
	at anorm.SimpleSql.as(SimpleSql.scala:6) ~[zipkin-query.jar:na]
	at com.twitter.zipkin.storage.anormdb.AnormDependencyStore$$anonfun$getDependencies$1.apply(AnormDependencyStore.scala:57) ~[zipkin-query.jar:na]
	at com.twitter.zipkin.storage.anormdb.AnormDependencyStore$$anonfun$getDependencies$1.apply(AnormDependencyStore.scala:32) ~[zipkin-query.jar:na]
	at com.twitter.zipkin.storage.anormdb.DB.withRecoverableRetry(DB.scala:120) ~[zipkin-query.jar:na]
	at com.twitter.zipkin.storage.anormdb.DB$$anonfun$inNewThreadWithRecoverableRetry$1.apply(DB.scala:147) ~[zipkin-query.jar:na]
	at com.twitter.util.Try$.apply(Try.scala:13) ~[zipkin-query.jar:na]
	at com.twitter.util.ExecutorServiceFuturePool$$anon$2.run(FuturePool.scala:115) [zipkin-query.jar:na]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_66]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_66]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_66]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_66]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_66]
Caused by: org.mariadb.jdbc.internal.util.dao.QueryException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')
AND a_key in ("sr","sa")
AND endpoint_service_name is not null
GROUP BY trace_' at line 1
Query is:
SELECT DISTINCT trace_id, span_id, endpoint_service_name
FROM zipkin_annotations
WHERE trace_id IN ()
AND a_key in ("sr","sa")
AND endpoint_service_name is not null
GROUP BY trace_id, span_id
        
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:478) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.result(AbstractQueryProtocol.java:399) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:381) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:337) ~[zipkin-query.jar:na]
	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:280) ~[zipkin-query.jar:na]
	... 37 common frames omitted
23:39:45.251 [pool-1-thread-2] INFO  c.t.f.h.filters.AccessLoggingFilter - 10.240.0.3 - - [14/Dec/2015:23:39:45 +0000] "GET /api/v1/dependencies?endTs=1450136385116 HTTP/1.1" 500 36 27 "-"
```
Data is present in Mysql tables from zipkin-query and other services. My dependency page is rendered blank. Is this expected behavior?

--- a/zipkin-anormdb/src/main/scala/com/twitter/zipkin/storage/anormdb/AnormDependencyStore.scala
+++ b/zipkin-anormdb/src/main/scala/com/twitter/zipkin/storage/anormdb/AnormDependencyStore.scala
@@ -46,27 +46,31 @@ case class AnormDependencyStore(val db: DB,
           case traceId ~ parentId ~ id => (traceId, parentId, id)
         }) *).groupBy(_._1)
 
-      val traceSpanServiceName: Map[(Long, Long), String] = SQL(
-        """SELECT DISTINCT trace_id, span_id, endpoint_service_name
-          |FROM zipkin_annotations
-          |WHERE trace_id IN (%s)
-          |AND a_key in ("sr","sa")
-          |AND endpoint_service_name is not null
-          |GROUP BY trace_id, span_id
-        """.stripMargin.format(parentChild.keys.mkString(",")))
+      if (parentChild.isEmpty) {
+        Seq.empty[DependencyLink]
+      } else {
+        val traceSpanServiceName: Map[(Long, Long), String] = SQL(
+          """SELECT DISTINCT trace_id, span_id, endpoint_service_name
+            |FROM zipkin_annotations
+            |WHERE trace_id IN (%s)
+            |AND a_key in ("sr","sa")
+            |AND endpoint_service_name is not null
+            |GROUP BY trace_id, span_id
+          """.stripMargin.format(parentChild.keys.mkString(",")))
         .as((long("trace_id") ~ long("span_id") ~ str("endpoint_service_name") map {
           case traceId ~ spanId ~ serviceName => (traceId, spanId, serviceName)
         }) *).map(r => (r._1, r._2) -> r._3).toMap
 
-      parentChild.values.flatMap(identity).flatMap(r => {
-        // parent can be empty if a root span is missing
-        for (
-          parent <- traceSpanServiceName.get((r._1, r._2));
-          child <- traceSpanServiceName.get((r._1, r._3))
-        ) yield (parent, child)
-      })
-      .groupBy(identity).mapValues(_.size) // sum span count
-      .map{ case ((parent, child), count) => DependencyLink(parent, child, count)}.toSeq
+        parentChild.values.flatMap(identity).flatMap(r => {
+          // parent can be empty if a root span is missing
+          for (
+            parent <- traceSpanServiceName.get((r._1, r._2));
+            child <- traceSpanServiceName.get((r._1, r._3))
+          ) yield (parent, child)
+        })
+        .groupBy(identity).mapValues(_.size) // sum span count
+        .map{ case ((parent, child), count) => DependencyLink(parent, child, count)}.toSeq
+      }
     } finally {
       returnConn(conn, borrowTime, "getDependencies")
     }

--- a/zipkin-common/src/test/scala/com/twitter/zipkin/storage/DependencyStoreSpec.scala
+++ b/zipkin-common/src/test/scala/com/twitter/zipkin/storage/DependencyStoreSpec.scala
@@ -79,6 +79,11 @@ abstract class DependencyStoreSpec extends JUnitSuite with Matchers {
     result(store.getDependencies(today + 1000)) should be(dep.links)
   }
 
+  /** Edge-case when there are no spans, or instrumentation isn't logging annotations properly. */
+  @Test def getDependencies_empty() = {
+    result(store.getDependencies(today + 1000)) should be(Seq.empty)
+  }
+
   /**
    * Trace id is not required to be a span id. For example, some instrumentation may create separate
    * trace ids to help with collisions, or to encode information about the origin. This test makes

